pkg/config/datafileprojectconfig/config.go

```diff
@@ -161,6 +169,16 @@ func (c DatafileProjectConfig) GetGroupByID(groupID string) (entities.Group, err
 	return entities.Group{}, fmt.Errorf(`group with ID "%s" not found`, groupID)
 }
 
+// GetOptimizelyConfig is the main interface to get OptimizelyConfig object
+func (c DatafileProjectConfig) GetOptimizelyConfig() *entities.OptimizelyConfig {
+
+	featuresList := c.GetFeatureList()
+	experimentsList := c.GetExperimentList()
+	revision := c.GetRevision()
+
+	return entities.NewOptimizelyConfig(featuresList, experimentsList, revision)
```

Comment:
@jaeopt should we be caching this lazily instead of re-computing every time? This is tied directly to the project config so we don't have to worry about the cached version becoming stale.

Reply:
my understanding is that we want to rebuild it every time for now since keeping that in memory will be expensive. But @jaeopt  can confirm.

Reply:
Yes we discussed the issue in a slack thread. We'll let client cache it for now and will visit again later to see the optimization is needed in the SDK level.